### ISSUE TITLE
1.4.11 Understanding: Additional focus indication of text inputs (default caret visible)

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/wcag#1014]
The Understanding doc for 1.4.11 "Non-text contrast" has an example in [section "Active User Interface Component Examples](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast#active-user-interface-component-examples)" > "Text input focus style":  a text field with additional focus indication.

It says:

> "Text input focus style: A focus indicator is required. If the focus indicator is styled by the author, it must meet the 3:1 contrast ratio."

As @patrickhlauke has pointed out in https://github.com/w3c/wcag/issues/680#issuecomment-484517551 

> But the image there shows the text input, and the focus outline on that input yes...but there's a caret also shown. This seems to then suggest that yes, regardless of caret, the outline must have sufficient 3:1 contrast. But if that outline is secondary, and the state of the input (it being focused) is already indicated by the presence of the cursor/caret...then it's irrelevant whether or not that outline is contrasty enough, no?

I believe this issue has not been resolved. The current example would see a site that uses **only** the cursor without **any** additional focus highlight PASS while an implementation with additional highlight (but below 3:1) fails. This seems counter-intuitive to me.

Options:
IMO, either we say "cursor alone is never enough to show focus", or additional focus indications should pass whenever there is already one / any other form of contrasty focus indication (e.g., a blinking caret).

**Comment by detlevhfischer** [comment]
@alastc @patrickhlauke  
This is still open, the example still sits under "[Active User Interface Component Examples](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast#active-user-interface-component-examples) >Text input focus style".
One proposal would be to say clearly that a caret on is own is not sufficient as focus indicator. But if it is deemed sufficient minimally, highlighting the fieldadditionally is only a best practice, but no requirement?

**Comment by patrickhlauke** [comment]
i've lost track of the various tidbits where i've pointed at gaps/inaccuracies, admittedly (and i have low energy at the moment for further extended discussions and wrestling matches on these). but: only going by the old Focus Visible, I'd argue that the caret itself is a sufficient indication of focus. any outline is then additional to that control. so i'd clearly say that in that case, the CARET is the primary focus indication, and that it passes Focus Visible (hence 1.4.11 does not apply to the outline, since that's secondary, and even if it wasn't there when the input has focus, it would be "visible" which input currently has focus - note that this does NOT apply to non-focused inputs...if there's no current focus on the input, and its only visible presence is the outline, or a bottom border, then yes that IS necessary to be able to perceive the presence of the input, so contrast ratio requirements apply there). then, mop up anything else under Focus Visible (Enhanced)

**Comment by patrickhlauke** [comment]
i would say, based on the above, that the entire example of "Text input focus style" - the entire row there in that table - should be removed. as arguably the caret is the focus indication.

**Comment by patrickhlauke** [comment]
> I believe this issue has not been resolved. The current example would see a site that uses only the cursor without any additional focus highlight PASS while an implementation with additional highlight (but below 3:1) fails. This seems counter-intuitive to me.

would the second case fail? i wouldn't...assuming the second case still has the caret. that's the main visible focus indication. the outline is window dressing (under strict reading of Focus Visible and Non-Text Content)

**Comment by patrickhlauke** [comment]
(and if the focus indicator is the caret, it would only fail if the author styled THE FOCUS INDICATOR, which would be the cursor...can authors even influence that? i'm guessing using the text color property perhaps?)

**Comment by mraccess77** [comment]
I'd say the caret provided by the browser that is unchanged by the color would pass this SC as currently written.  I agree with @patrickhlauke that the understanding should be updated with the input examples removed/changed.  Focus state on a checkbox or select is probably a better example as there caret.

**Comment by patrickhlauke** [comment]
if we're agreed that the caret does count as sufficient focus indication, thus 1.4.11 not applying to the additional outline, may be good to even explicitly state that particular case/question

**Comment by alastc** [comment]
Ok, changed from "discussion" to "needs assigning", i.e. need a PR to show the change and survey.

**Comment by patrickhlauke** [comment]
I'll have a stab at a PR later today...

**Comment by patrickhlauke** [comment]
made a first stab here #1065

**Comment by mbgower** [comment]
The fact that every default user agent implementation of an input I have ever seen includes a focus indicator beyond the flashing caret, suggests to me that the insertion caret's primary role is an indicator of the location of the input, not an indicator of focus.
Further, if I tab to a prepopulated input, there IS no insertion caret.
The consistent indicator of focus in both these scenarios is the indictor around the text box.
If the cursor was not flashing, would anyone consider it a sufficient indicator of location? Since movement is not captured as a criterion for focus indication, extending the logic would suggest a non-flashing cursor should also be sufficient.

So I get the argument, but I have difficulty believing that anyone in this thread would be happy with a site where there is no indicator around an input field beyond the cursor.

**Comment by patrickhlauke** [comment]
> the insertion caret's primary role is as an indicator of the location of the input, not an indicator of focus

but insertion happens (for standard controls at least) on the same element that has focus, so the distinction is moot? unless it's a situation where insertion point and focused element are different (but in that case, if the focused element has no indication, and there's only a caret, then the former bit would fail 2.4.7 already?)

> Further, if I tab to a prepopulated input, there IS no insertion caret.

but instead the prepopulated text gets highlighted/selected.

> If the cursor was not flashing, would anyone consider it a sufficient indicator of location? Since movement is not captured as a criterion for focus indication, extending the logic would suggest a non-flashing cursor should also be sufficient.

per 2.4.7's very undefined definition of "visible", the caret would - in my view at least - pass the criterion as it is unarguably "visible", particularly when *not* flashing (it was the flashing nature of the caret that cast more doubt on this, as it's then not "visible" all the time).

>  I have difficulty believing that anyone in this thread would be happy with a site where there is no indicator beyond the cursor for an input field with focus

not "happy", but "resigned to pass this based on the wording of the success criteria" (when it comes to 2.4.7 and 1.4.11)

**Comment by mraccess77** [comment]
I have seen many input fields with only carets -- in some browsers the default focus change is also so faint I fail to detect it as it's a change of contrast on the border (IE) it but I do see the caret.  In Firefox the change of focus on input is on the border as well and without seeing hue it would be very difficult to discern.    Under the new proposed enhanced visual focus criteria a narrow caret would fail and a faint contrast change would fail.
